sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -305,6 +321,40 @@ static void _init_sub_prefix(ipv6_addr_t *out,
     out->u8[bytes] |= idx << shift;
 }
 
+static void _init_sub_prefix_uid(ipv6_addr_t *out,
```

Comment:
This function maybe becomes simpler when using the 64bit integer member in the IPv6 address.
Init prefix -> convert `uid` to 64 bit integer -> shift that by `bits` -> `OR` prefix 64bit integer with `uid` 64 bit integer.

Reply:
And under the assumption that hw addresses of the same manufacturer differ most in the least significant bits, it might be worth to prefer those.